tools/common/Application.cs

```diff
@@ -132,6 +140,37 @@ public partial class Application
 				}
 			}
 		}
+
+		public bool IsDeviceBuild {
+			get {
+				switch (Platform) {
+				case ApplePlatform.iOS:
+				case ApplePlatform.TVOS:
+				case ApplePlatform.WatchOS:
+					return BuildTarget == BuildTarget.Device;
+				case ApplePlatform.MacOSX:
+					return false;
+				default:
+					throw ErrorHelper.CreateError (71, Errors.MX0071, Platform, ProductName);
+				}
+			}
+		}
```

Comment:
Merging mtouch/mmp code is nice when it reduce the total amount of code.

However if every member has to switch on the platform then it's not a gain, in size or in readability. Adding a new platform also means changing code all over the place.

Why not just use some polymorphism ? `MacApplication : Application` and `iOSApplication : Application`.

It was almost the case with the partial classes and custom files, except they required different binaries. OTOH subclasses would not prevent both/all classes to be in the same binary (if that's your end goal).